### ISSUE TITLE
combine i3ipc with other event loops

### ISSUE THREAD

**Issue by ghistes** [altdesktop/i3ipc-python#134]
Hi,

is it possible (and if it is how do you do it) to combine the event-loop from i3ipc with another event loop (gtk say) so that you could for example write a GUI-script that uses i3ipc?

Many thanks!

**Comment by acrisci** [comment]
No, it's not currently possible. But I would like to support the use case and I'm open to a discussion about how to implement it.

**Comment by acrisci** [comment]
For GTK in particular, I have [this project](https://github.com/altdesktop/i3ipc-glib/issues) but I haven't updated in awhile.

**Comment by ghistes** [comment]
I can't really contribute to any discussion as I am pretty new to Python and don't know the ecosystem.
But it maybe worth you can get some ideas from Perl's AnyEvent-module (https://metacpan.org/pod/AnyEvent).

**Comment by Aphray** [comment]
I realize this is a slightly old thread, however I felt I should add that it is possible to combine with the event loop of Qt using [asyncqt](https://github.com/gmarull/asyncqt). The snippet below achieves just that.

```
from i3ipc.aio import Connection
from i3ipc import Event
from PyQt5.QtWidgets import QApplication, QWidget
from asyncqt import QEventLoop
import sys
import asyncio

def _print(conn, event):
    print(event)

async def main():
    i3 = await Connection(auto_reconnect=True).connect()
    i3.on(Event.WINDOW, _print)
    await i3.main()

if __name__ == '__main__':
    app = QApplication(sys.argv)
    loop = QEventLoop(app)
    asyncio.set_event_loop(loop)
    widget = QWidget()
    widget.show()
    with loop:
        sys.exit(loop.run_until_complete(main()))
```

In case anyone is looking to combine with Qt.
